tests/lib/util/update-metadata-builder-test.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * @fileoverview Tests for UpdateMetadataBuilder
+ */
+
+'use strict';
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+var assert = require('chai').assert,
+	UpdateMetadataBuilder = require('../../../lib/util/update-metadata-builder');
+
+// ------------------------------------------------------------------------------
+// Tests
+// ------------------------------------------------------------------------------
+
+describe('UpdateMetadataBuilder', function() {
+
+	var builder;
+
+	beforeEach(function() {
+		builder = new UpdateMetadataBuilder();
+	});
+
+	describe('constructor', function() {
+
+		it('should create empty body', function() {
+			var body = builder.getRequestBody();
+			assert.equal(body.length, 0);
+		});
+	});
+
+	describe('insertPath()', function() {
+		it('should throw when op is null', function() {
+			assert.throws(function() {
+				builder.insertPath(null, '/test', '10');
+			});
+		});
+
+		it('should throw when op is empty', function() {
+			assert.throws(function() {
+				builder.insertPath('', '/test', '10');
+			});
+		});
+
+		it('should throw when op is not a string', function() {
+			assert.throws(function() {
+				builder.insertPath('', '/test', '10');
+			});
+		});
+
+		it('should throw when path is null', function() {
+			assert.throws(function() {
+				builder.insertPath('add', null, '10');
+			});
+		});
+
+		it('should throw when path is empty', function() {
+			assert.throws(function() {
+				builder.insertPath('add', '', '10');
+			});
+		});
+
+		it('should throw when path is not a string', function() {
+			assert.throws(function() {
+				builder.insertPath('add', 10, '10');
+			});
+		});
+
+		it('should throw when value is undefined', function() {
+			assert.throws(function() {
+				builder.insertPath('add', 10);
+			});
+		});
+	});
+
+	describe('insertFrom()', function() {
```

Comment:
These tests only test the failure cases; we should also have unit tests to ensure that the methods do what they should in the happy path.